tests/core/arch-x86/stalker-x86-exceptions.cpp

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright (C) 2009-2021 Ole André Vadla Ravnås <oleavr@nowsecure.com>
+ * Copyright (C) 2010-2013 Karl Trygve Kalleberg <karltk@boblycat.org>
+ *
+ * Licence: wxWindows Library Licence, Version 3.1
+ */
+#include <glib.h>
+
+extern "C" {
+  void test_check_bit (guint32 * val, guint8 bit);
+  void test_set_bit (const char * func, guint32 * val, guint8 bit);
+  void test_try_and_catch (guint32 * val);
+  void test_try_and_dont_catch (guint32 * val);
+}
+
+class TestException;
+class FakeException;
+class TestResource;
+
+static void test_try_and_catch_pp (guint32 * val);
+static void test_try_and_dont_catch_pp (guint32 * val);
+static void test_try_and_dont_catch_pp_2 (guint32 * val);
+static void test_try_and_dont_catch_pp_3 (guint32 * val);
+
+extern "C" {
+
+  void
+  test_check_bit (guint32 * val,
+                  guint8 bit)
+  {
+    g_assert_true (*val & ((guint32) 1 << bit));
+    *val = *val & ~((guint32) 1 << bit);
+  }
+
+  void
+  test_set_bit (const char * func,
+                guint32 * val,
+                guint8 bit)
+  {
+    if (g_test_verbose ())
+      g_print ("\tFunc: %s, Set: %d\n", func, bit);
+
+    *val = (*val | ((guint32) 1 << bit));
+  }
+
+  void
+  test_try_and_catch (guint32 * val)
+  {
+    test_try_and_catch_pp (val);
+  }
+
+  void
+  test_try_and_dont_catch (guint32 * val)
+  {
+    test_try_and_dont_catch_pp (val);
+  }
+
+}
+
+class TestException {
+
+};
+
+class FakeException {
+
+};
+
+class TestResource {
+  private:
+    guint32 * val;
+  public:
+    TestResource (guint32 * val) : val (val)
+    {
+      test_set_bit ("TestResource", val, 0);
+    }
+    ~TestResource () {
+      test_set_bit ("TestResource", val, 1);
+    }
+};
+
+static void
+test_try_and_catch_pp (guint32 * val)
+{
+  try {
+
+    test_set_bit ("test_try_and_catch_pp", val, 0);
+
+    throw TestException ();
+
+    test_set_bit ("test_try_and_catch_pp", val, 1);
+  } catch (TestException &ex) {
```

Comment:
```suggestion
  } catch (TestException & ex) {
```

Space after ampersand, just like asterisk. Here and elsewhere.